### PR TITLE
feat(VETS-CPC- 752) get the average ratings of a specific vet

### PR DESCRIPTION
**Pull request by DenisaNicoletaH**
###  **JIRA:** 
champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/35?modal=detail&selectedIssue=CPC-752&sprint=293

### **Context:**
This user story is about showing the average rating that a each vet will have depending on the ratings that they receive from the users

### **Changes:**
- Implementation of GET the average rating for each vet in the Vet-Service
- Implementation of GET the average rating for each vet in the API-Gateway
- 90% jacoco coverage + testing in Vet-Service
- 90% jacoco coverage + testing in API-Gateway

###  **Before and After UI (Required for UI-impacting PRs)**
image
image

### CODE REVIEW

**Review by ayasophya — DISMISSED** [review]
everything implemented seems to work fine

**Review by Sprytte — DISMISSED** [review]
All looks good! The acceptance criteria for the story are met and the tests were properly implemented.

File: vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerIntegrationTest.java [hunk 1]
```diff
@@ -155,6 +155,45 @@ void deleteARatingForVet_WithValidId_ShouldSucceed() {
...
+                .get()
+                .uri("/vets/" + INVALID_VET_ID + "/ratings/average")
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(Double.class)
+                .value(avg ->{
+                            assertEquals(0.0,avg);
+                        }
+                );
```
Comment (DRMADNESS2004) [inline vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerIntegrationTest.java:194]: Controller Integration test is reactive, looks right. Just the test with invalid vetId could return a NotFound exception.

File: vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerUnitTest.java [hunk 2]
```diff
@@ -128,6 +128,37 @@ void getNumberOfRatingsByVetId_ShouldSucceed() {
...
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody(Double.class)
+                .value(d ->{
+                    assertEquals(d,averageRating);
+                });
+
+        Mockito.verify(ratingService, times(1))
+                .getAverageRatingByVetId(ratingDTO.getVetId());
+
+    }
```
Comment (DRMADNESS2004) [inline vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerUnitTest.java:159]: Controller Unit test is reactive, maybe the lambda parameter 'd' could be named to something more relevant, 'a' for "average rating" for example or "actualAverageRating".

**Review by DRMADNESS2004 — DISMISSED** [review]

File: vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerIntegrationTest.java [hunk 3]
```diff
@@ -155,6 +155,44 @@ void deleteARatingForVet_WithValidId_ShouldSucceed() {
...
+                .get()
+                .uri("/vets/" + ratingRequestDTO.getVetId() + "/ratings/average")
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(Double.class)
+                .value(avg ->{
+                            assertEquals(0.0,avg);
+                        });
+    }
```
Comment (DRMADNESS2004) [inline vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerIntegrationTest.java:194]: Controller Integration tests are reactive, seem valid.

Comment (DRMADNESS2004) [inline vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerIntegrationTest.java:194]: Controller Integration tests are reactive, follows naming convention, seem valid.

**Review by DRMADNESS2004 — APPROVED** [review]
Overall, tests are well-done, code looks clean and follows naming conventions, code is reactive, front-end looks good, everything seems to work

**Review by Mila5847 — DISMISSED** [review]
The front-end looks good, the code is tested, and the acceptance criteria are met.

**Review by KarinaSofia — APPROVED** [review]
Effective changes to the ApiGateway and VetController tests.

**Review by Mila5847 — APPROVED** [review]
Tests were updated.

**Pull request merged**